code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DefaultDeploymentUnitInstance.java

```diff
@@ -26,17 +30,29 @@
 import io.cattle.platform.servicediscovery.deployment.DeploymentUnitInstance;
 import io.cattle.platform.servicediscovery.deployment.InstanceUnit;
 import io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.DeploymentServiceContext;
+import io.cattle.platform.util.exception.InstanceException;
 import io.cattle.platform.util.exception.ServiceInstanceAllocateException;
 
 import java.util.Arrays;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang3.tuple.Pair;
 
 public class DefaultDeploymentUnitInstance extends DeploymentUnitInstance implements InstanceUnit {
+    private static final Set<String> ERROR_STATES = new HashSet<String>(Arrays.asList(
+            InstanceConstants.STATE_ERRORING,
+            InstanceConstants.STATE_ERROR));
+    private static final Set<String> BAD_ALLOCATING_STATES = new HashSet<String>(Arrays.asList(
+            InstanceConstants.STATE_ERRORING,
+            InstanceConstants.STATE_ERROR,
+            InstanceConstants.STATE_STOPPING,
```

Comment:
state_stopping and state_stopped wouldn't be bad allocating state for service with start_once=true. We should probably check start_count flag along with checking the stopped/stopping state